doc/ipythonconsole.rst

```diff
@@ -2,91 +2,159 @@
+
+You can also change the name of the current :guilabel:`IPython console` tab with the :guilabel:`Rename tab` option, or by simply double-clicking it.
+
+.. image:: images/console/console-rename.gif
+   :alt: Spyder IPython Console showing renaming console
+
+
+
+.. _connecting-external-kernel:
+
+==================================
+Using external kernels with Spyder
+==================================
+
+You can connect to external local and remote kernels (including those managed by Jupyter Notebook or QtConsole) through the :guilabel:`Connect to an existing kernel` dialog under the :guilabel:`Consoles` menu.
+For this feature to work, a compatible version of the ``spyder-kernels`` package :ref:`must be installed <starting-kernel-problems-ref>` in the environment or machine in which the external kernel is running.
 
 .. image:: images/console/console-menu.png
-   :alt: Spyder IPython Console as above, with the options menu open
+   :alt: Connect to external kernel dialog of the Spyder IPython console
 
 
-Connect to an external kernel
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Connect to a local kernel
+~~~~~~~~~~~~~~~~~~~~~~~~~
 
-.. note::
-   If on Windows and connecting to a remote machine over ``ssh``, you'll need to install the ``paramiko`` python package first, *e.g.* with ``conda install paramiko`` if using Anaconda.
+To connect to a local kernel that is already running (e.g. one started by Jupyter notebook),
 
-To connect to an external kernel,
+#. Run ``%connect_info`` in the notebook or console you want to connect to, and copy the name of its kernel connection file, shown after ``jupyter <app> --existing``.
 
-#. Launch an IPython kernel on the local or remote host if one is not already running.
+   .. image:: images/console/console-connect-local-step1.gif
+      :alt: Running connect_info in a Jupyter notebook
 
-   * If using Spyder 3.3.0 or later, you'll need to launch the kernel with ``python -m spyder_kernels.console`` (after you've first installed ``spyder-kernels`` on the host with ``<conda/pip> install spyder-kernels``).
-   * If using a version of Spyder before 3.3.0, ``ipython kernel`` should work to launch the kernel, albeit without certain Spyder-specific features.
+#. In Spyder, click :guilabel:`Connect to an existing kernel` from the :guilabel:`Consoles` menu, and paste the name of the :guilabel:`Connection file` from the previous step.
 
-#. Copy the connection file (:file:`{jupyter/runtime/dir/path}/kernel-{pid}.json`) to the machine you're running Spyder on (if remote) or note its location (if local).
+   As a convenience, kernel ID numbers (e.g. ``1234``) entered in the connection file path field will be expanded to the full path of the file, i.e. :file:`{jupyter/runtime/dir/path}/kernal-{id}.json`.
 
-   You can get :file:`{jupyter/runtime/dir/path}` by executing ``jupyter --runtime-dir`` in the same Python environment as the kernel.
+   .. image:: images/console/console-connect-local-step2.gif
```

Comment:
Since it's not possible to see what menu entry you're clicking at the beginning of the gif to bring the `Connect to kernel` dialog, I suggest to film this gif with a much smaller screen resolution (for instance, the one @juanis2112 used for the hero screenshot of this pane).

That would also avoid the time spent in zooming in to show the dialog.

Reply:
If I changed the screen resolution I would need to redo all the GIFs to match, which isn't possible right now as I'm on a trip visiting family and don't have access to the machine I set up to run a remote kernel on. However, I can edit it to zoom in at the beginning before it shows the Consoles menu, so its easily readable, and then just pan across to show the kernel dialog, and I can crank up the speed of the zooms/moves and increase the frame rate to match, so they take up less time but still look smooth.

Reply:
Actually, I do have a Pi with me, so given a bit of time I think I can get it set up to work, if its worth doing every GIF again.

Reply:
Also, to note, @juanis2112 actually has a much higher resolution than I do because she has a HiDPI screen, the UI is just scaled via macOS to appear as if it were on a much smaller one. I can simulate something sort of similar on my Win 8.1 machine, so we don't have to use overly low source resolutions for the GIFs, but the UI doesn't look nearly as good on Windows and a number of things are not rendered well, so I think I just have to go with the true 720p option:

UI scaled to max via OS:

![image](https://user-images.githubusercontent.com/17051931/103042539-fd52f900-4547-11eb-85f0-0c3a5d3aceda.png)

Monitor downsampled to 1280 x 720:

![image](https://user-images.githubusercontent.com/17051931/103042645-5589fb00-4548-11eb-9667-1b081df03430.png)

Reply:
I redid this as specified above, which despite a long wait I didn't receive feedback otherwise on, so this should be addressed for now